File: MasteringDart/lib/chapters/chapter08/basic/basic_classes.dart
// ignore_for_file: avoid_print
//
//  basic_classes.dart
//  MasteringDart
//
//  Capítulo 08 — Clases y objetos.
//  Nivel BÁSICO: Clase simple con propiedades, constructor y creación de objetos.
//
//  Ejecutar con: dart run lib/chapters/chapter08/basic/basic_classes.dart

/// Clase User simple con dos propiedades: name y age.
/// El constructor usa this.name y this.age para asignación abreviada.
class User {
  final String name;
  final int age;

  User(this.name, this.age);
}

/// Ejecuta ejemplos básicos de clases.
void runBasicClassExamples() {
  // --- Crear objetos usando el constructor ---
  final user1 = User('Alice', 28);
  final user2 = User('Bob', 32);

  // --- Acceder a propiedades e imprimir ---
  print('User 1: ${user1.name}, ${user1.age} years old');
  print('User 2: ${user2.name}, ${user2.age} years old');

  // --- Imprimir el objeto ---
  print('user1 = $user1');
}

void main() {
  runBasicClassExamples();
}

File: MasteringSwift/MasteringSwift/chapters/chapter08/basic/BasicClasses.swift
//
//  BasicClasses.swift
//  MasteringSwift
//
//  Capítulo 08 — Clases y objetos.
//  Nivel BÁSICO: Clase simple con propiedades, inicializador y creación de objetos.
//

import Foundation

/// Clase User simple con dos propiedades: name y age.
fileprivate class User {
    let name: String
    let age: Int

    /// Inicializador designado (constructor).
    init(name: String, age: Int) {
        self.name = name
        self.age = age
    }
}

/// Ejecuta ejemplos básicos de clases. Llamar desde un test o playground.
func runBasicClassExamples() {
    // --- Crear objetos usando el inicializador ---
    let user1 = User(name: "Alice", age: 28)
    let user2 = User(name: "Bob", age: 32)

    // --- Acceder a propiedades e imprimir ---
    print("User 1: \(user1.name), \(user1.age) years old")
    print("User 2: \(user2.name), \(user2.age) years old")

    // --- Imprimir el objeto ---
    print("user1 = \(user1)")
}
